README.md

```diff
@@ -47,9 +47,9 @@ Let's dive into a simple use case to see how Stimulus Store works. In this examp
 
 ```js
 // controllers/stores/counter.js
-import { Store } from "stimulus-store"; // Import the shared Store class
+import { createStore } from "stimulus-store";
 
-export const counterStore = new Store(0); // Initialize with an initial value of 0
+export const counterStore = createStore(name: 'counterStore', initialValue: 0, type: Number)
```

Comment:
Question about this:
- Why can't it be achieved with generics + inference  vs recreating a type system?

Reply:
To enforce type safety at runtime when the store sets a value

https://github.com/omarluq/stimulus-store/blob/ae8b4cd75bcdf235c0aa232d3f3d1ad5706e480c/src/store.ts#L51-L60

And here using generics and interfaces would have made the API more complex and harder to use. I feel like this simple typing gives us all we need without overweight.
https://github.com/omarluq/stimulus-store/blob/ae8b4cd75bcdf235c0aa232d3f3d1ad5706e480c/src/createStore.ts#L34-L64